### DISCUSSION TITLE
Error running: dandi organize <source_folder> -f dry

### DISCUSSION THREAD

**Discussion by sumachinta** [dandi/helpdesk#80] [Q&A]
After validating nwb files in the source folder with no validation errors, unable to organize the files. 
(dandi) C:\Users\Pluta lab\000301>dandi organize "D:\MATLAB Analysis\NWD\matnwb\nwd_files" -f dry
2022-08-30 12:09:26,450 [    INFO] Loading metadata from 14 files
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 12 concurrent workers.
[Parallel(n_jobs=-1)]: Done   3 out of  14 | elapsed:    4.3s remaining:   16.1s
[Parallel(n_jobs=-1)]: Done   5 out of  14 | elapsed:    4.4s remaining:    8.0s
[Parallel(n_jobs=-1)]: Done   7 out of  14 | elapsed:    4.5s remaining:    4.5s
[Parallel(n_jobs=-1)]: Done   9 out of  14 | elapsed:    4.5s remaining:    2.5s
[Parallel(n_jobs=-1)]: Done  11 out of  14 | elapsed:    4.6s remaining:    1.2s
[Parallel(n_jobs=-1)]: Done  14 out of  14 | elapsed:    4.6s finished
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_26.nwb -> sub-M26\sub-M26_ses-20200319T153237_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_27.nwb -> sub-M27\sub-M27_ses-20200319T163241_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_29.nwb -> sub-M29\sub-M29_ses-20200612T143309_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_36.nwb -> sub-M36\sub-M36_ses-20200827T141103_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_37.nwb -> sub-M37\sub-M37_ses-20200829T154548_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_38.nwb -> sub-M38\sub-M38_ses-20200904T142415_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_40.nwb -> sub-M40\sub-M40_ses-20201126T145724_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_41.nwb -> sub-M41\sub-M41_ses-20201127T144127_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_42.nwb -> sub-M42\sub-M42_ses-20201201T143600_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_43.nwb -> sub-M43\sub-M43_ses-20201203T150431_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_44.nwb -> sub-M44\sub-M44_ses-20201207T155850_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_45.nwb -> sub-M45\sub-M45_ses-20201209T141009_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_50.nwb -> sub-M50\sub-M50_ses-20211026T152836_behavior+ecephys.nwb
DRY: D:\MATLAB Analysis\NWD\matnwb\nwd_files\ecephys_t_52.nwb -> sub-M52\sub-M52_ses-20211102T155806_behavior+ecephys.nwb
2022-08-30 12:09:31,192 [    INFO] Organized 0 out of 14 paths. Visit ./
2022-08-30 12:09:31,192 [    INFO] Logs saved in C:\Users\Pluta lab\AppData\Local\dandi\dandi-cli\Logs\20220830160919Z-2440.log

**Comment by CodyCBakerPhD** [comment]
@sumachinta What happens if you remove the `-f dry` on the end (*i.e.*, moving to the next step of 2e from https://www.dandiarchive.org/handbook/13_upload/#data-uploadmanagement-workflow)?

Does it print out an error in the console (and if so, can you copy/paste that error here?)

**Reply by sumachinta** [reply]
Hi Dr. Baker, 
Thank you for the quick response.
When I run: dandi organize "D:\MATLAB Analysis\NWD\matnwb\nwd_files", I get the following [WinError 17]

(dandi) C:\Users\Pluta lab\000301>dandi organize "D:\MATLAB Analysis\NWD\matnwb\nwd_files"
2022-08-30 13:10:28,635 [    INFO] Loading metadata from 14 files
[Parallel(n_jobs=-1)]: Using backend LokyBackend with 12 concurrent workers.
[Parallel(n_jobs=-1)]: Done   3 out of  14 | elapsed:    4.4s remaining:   16.3s
[Parallel(n_jobs=-1)]: Done   5 out of  14 | elapsed:    4.4s remaining:    8.0s
[Parallel(n_jobs=-1)]: Done   7 out of  14 | elapsed:    4.4s remaining:    4.4s
[Parallel(n_jobs=-1)]: Done   9 out of  14 | elapsed:    4.4s remaining:    2.4s
[Parallel(n_jobs=-1)]: Done  11 out of  14 | elapsed:    4.4s remaining:    1.1s
[Parallel(n_jobs=-1)]: Done  14 out of  14 | elapsed:    4.5s finished
2022-08-30 13:10:33,207 [    INFO] Hard link support autodetected; setting files_mode='hardlink'
2022-08-30 13:10:33,230 [    INFO] Logs saved in C:\Users\Pluta lab\AppData\Local\dandi\dandi-cli\Logs\20220830171021Z-9648.log
Error: [WinError 17] The system cannot move the file to a different disk drive: 'D:\\MATLAB Analysis\\NWD\\matnwb\\nwd_files\\ecephys_t_26.nwb' -> '.\\sub-M26\\sub-M26_ses-20200319T153237_behavior+ecephys.nwb'

**Reply by CodyCBakerPhD** [reply]
Hmm... interesting, haven't seen this one before - @yarikoptic what would you suggest here; perhaps trying with `-f symlink`? Would that resolve cross-disk drive permissions?

What I would try otherwise is redoing step 2e from the beginning (_i.e._, redo the `dandi download` step) starting from the same disk drive that the NWB files are on ("D:") and call `dandi organize <relative_path_to_nwb_folder>` as a relative path so there's no cross-disk issues.

**Reply by CodyCBakerPhD** [reply]
@sumachinta If you still experience issues after those two suggestions (or any @yarikoptic might have), feel free to schedule a time on my [calendly](https://calendly.com/cody-c-baker-phd) and I'll be happy to help try to resolve this one-on-one

**Reply by yarikoptic** [reply]
filed https://github.com/dandi/dandi-cli/issues/1110 to get it resolved "properly". Meanwhile, since you are operating across drives anyways (so not like you would run out of space this way but not with `move`) you can just use `organize --files-mode copy`.

**Reply by sumachinta** [reply]
@CodyCBakerPhD @yarikoptic I tried to redo step 2e from the beginning starting from the same disk drive on which the NWB files are ("D:"). This resolved the issue for me and all the files were uploaded successfully without any problems. 
Thank you so much for the suggestions.
